File: frontend/src/types/communication-log.ts
export type CommunicationMethod = "Meeting" | "Email" | "Phone" | "Chat";

export interface CommunicationLog {
  id: string;
  riskId: string;
  date: string;
  method: CommunicationMethod;
  stakeholder: string;
  notes: string;
  createdBy: string;
  createdByName: string;
  createdAt: string;
}

export interface CreateCommunicationLogInput {
  date: string;
  method: CommunicationMethod;
  stakeholder: string;
  notes: string;
}

export const COMMUNICATION_METHODS: CommunicationMethod[] = ["Meeting", "Email", "Phone", "Chat"];

export const COMMUNICATION_METHOD_LABELS: Record<CommunicationMethod, string> = {
  Meeting: "Meeting",
  Email: "Email",
  Phone: "Telepon",
  Chat: "Chat/Pesan",
};

File: backend/internal/domain/entity/communication_log.go
package entity

import (
	"time"

	"github.com/google/uuid"
)

// CommunicationLog represents communication records related to a risk.
type CommunicationLog struct {
	ID            uuid.UUID
	RiskID        uuid.UUID
	Date          time.Time
	Method        string // "Meeting", "Email", "Phone", "Chat"
	Stakeholder   string
	Notes         string
	CreatedBy     uuid.UUID
	CreatedByName string
	CreatedAt     time.Time
}

// CommunicationMethod constants
const (
	CommunicationMethodMeeting = "Meeting"
	CommunicationMethodEmail   = "Email"
	CommunicationMethodPhone   = "Phone"
	CommunicationMethodChat    = "Chat"
)

// ValidCommunicationMethods returns all valid communication methods
func ValidCommunicationMethods() []string {
	return []string{
		CommunicationMethodMeeting,
		CommunicationMethodEmail,
		CommunicationMethodPhone,
		CommunicationMethodChat,
	}
}

// IsValidMethod checks if the method is valid
func IsValidMethod(method string) bool {
	for _, m := range ValidCommunicationMethods() {
		if m == method {
			return true
		}
	}
	return false
}
